### DISCUSSION TITLE
prompt got clear sometimes, not sure why

### DISCUSSION THREAD

**Discussion by tuananh** [ghostty-org/ghostty#3476] [Q&A] [answered]
sometimes it becomes like this

![Image](https://github.com/user-attachments/assets/28768db2-8c8e-4fce-8244-2b3c0968381e)

when it supposes to look like this 

![Image](https://github.com/user-attachments/assets/bf49575d-0dde-46ef-99e6-bc2c0c6ea33b)

the config

```
theme = catppuccin-frappe
shell-integration = bash
font-family = Cascadia Mono NF
font-size = 11.5
cursor-style = bar
cursor-style-blink = true

# macOS only feature !!!
# keybind = global:super+t=toggle_quick_terminal
# quick-terminal-animation-duration = 0.1

window-decoration = false
copy-on-select = true
auto-update = off
```

**Comment by mitchellh** [comment]
Is this triggered by resizing or any specific action, or the prompt doesn't show up at all? I suspect this is a shell config issue but still want to figure Ito ut.

**Reply by tuananh** [reply]
Could be. Im using i3 tiling window manager so when i create a new windows, the old one will be resized. the prompt appears when i started typing (enter for example)

**Reply by 0x4a61636f62** [reply]
Same thing for me. The prompt is not visible on startup on **Gnome and Hyprland**. However, when I start typing, the prompt appears. It happens without resizing and regardless of a new installation without configuration.

![prompt-not-visible-on-startup](https://github.com/user-attachments/assets/770f1dc3-a37a-404c-89c5-e7876c8ce9fd)

I am on 1.0.0-3 and installed the package listed on  **`extra/ghostty`** on Arch. My shell is **Nushell**.

**Reply by tuananh** [reply]
oh yeah i forgot to mention, im also on arch, install from extra repo

**Reply by 0x4a61636f62** [reply]
I forgot to mention that it happens every time on startup. But after typing, the issue does not occur again, regardless of resizing etc.

**Comment by bac0n** [comment]
For me its any resize action, maximize (not minimize), snap, unsnap to screen edge or frame resize. but I need to press enter aleast once. It also seem to have something to do with two line prompt, doesn't happend if I convert it to a one line prompt (removing "\n").

prompt_command example:
```
PROMPT_COMMAND='PS1="\u@\h \w\n: "'
```

Edit:
![Screenshot_20241230_114118](https://github.com/user-attachments/assets/9e8283d2-8775-45e7-bb8f-11c932a1c787)

On the right split there's a newline in between, I split up, and split left (if I resize the window, the prompts that has atleast one enter will disappear).

**Comment by tuananh** [comment]
my guess is on 2 lines prompt too

**Comment by 0x4a61636f62** [comment]
@tuananh, @bac0n: Could you please check if you are also experiencing the following behavior: https://github.com/ghostty-org/ghostty/discussions/4780

For me, they are related. If my prompt "gets clear", then `xdg-open` no longer works.

**Comment by maximillianus** [comment]
Having similar issue, using Starship with multiline prompt here. When resizing, the upper lines are erased but reappear when I type enter to start prompt in newline. I have to say it does not happen everytime. For example, after I switch prompt to zsh and back to bash, the problem seems to disappear. It re-appears when I start new terminal window.

**Comment by 00-kat** [comment]
I've seen this issue before with some Nushell users, and disabling shell integration worked for them.  `shell-integration = none` if you don't use Nushell, and disable everything in `$env.config.shell_integration` if you do.  Not sure what the issue is.

**Reply by tuananh** [reply]
im too lazy to debug so i just change my prompt to 1 line. works fine since.

**Reply by maximillianus** [reply]
the shell-integration thing helps. thanks.

**Comment by emilia-miki** [comment]
I pinned down the issue to OSC133 handling. Enabling  it in nushell config is what makes ghostty not display the prompt on start. Works fine on alacritty and kitty, pretty sure should work in iterm2 as well. shell-integration is set to none, if that's relevant.

**Reply by 00-kat** [reply]
For the unaware, OSC 133 is the escape sequence for prompt marking.

**Comment by mitchellh** [comment, marked as answer]
I'm going to close this, there's no real reproduction or actionable outcome of this. If someone is experiencing this, let's open a new discussion with specific steps to reproduce and we can try to pinpoint a problem.
